Add restore procedure documentation (#130)
tembo-io/website: docs/tembo-cloud/backup-and-restore.md

- The retention policy for AWS S3 is set to 40 days
- AWS S3 object versioning is enabled, and the retention policy for non-current versions is 30 days

## WAL archiving

Tembo sets `archive_timeout` to 5min, ensuring that WAL files are closed and archived to S3 at least every five minutes, providing a deterministic time-based value for your Recovery Point Objective (RPO).

## Restore

Tembo Cloud supports Point-In-Time Recovery (PITR) to a new Tembo Instance for all running Instances and for 30 days after deletion.

:::info
Coming soon: Point-In-Time Recovery (PITR) via the [Tembo Cloud UI](https://cloud.tembo.io)
:::

> For information on authenticating to the API, please see the [Tembo Cloud API authentication guide](/docs/tembo-cloud/api-authentication)

Reviewer: ```suggestion
For information on authenticating to the API, please see the [Tembo Cloud API authentication guide](/docs/tembo-cloud/api-authentication).
```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>

> <TabItem value="curl" label="Curl">

Reviewer: since there is only 1 tab, maybe delete this stuff

```shell
export TEMBO_TOKEN=<your token>
export TEMBO_ORG=<your organization id>
export TEMBO_INST_RESTORE_FROM=<your instance id>
export TEMBO_INST_RESTORE_TO=<your new instance name>

> export TEMBO_RECOVERY_TIME=$(date -u +'%Y-%m-%dT%H:%M:%SZ' -d '-5 minutes')

Reviewer: I think export is not required

Reviewer: Please also add double quotes around the place to be inserted so they know what kind of quotes to use

Author: export is required if you don't run them all in a single command with curl.  Otherwise if you don't export it's not in your shell environment to use

Author: if we want to assume someone would run all the environment variables and the curl command in the same command then yes we can remove them.

> export TEMBO_RECOVERY_TIME=$(date -u +'%Y-%m-%dT%H:%M:%SZ' -d '-5 minutes')

Reviewer: Please add comment like 
```
# Recover to five minutes ago
```

curl -X 'POST' \
  "https://api.tembo.io/api/v1/orgs/$TEMBO_ORG/restore" \
  -H "accept: application/json" \
  -H "Authorization: Bearer $TEMBO_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{
    "instance_name": "$TEMBO_INST_RESTORE_TO",
    "restore": {
      "instance_id": "$TEMBO_INST_RESTORE_FROM",
      "recovery_target_time": "$TEMBO_RECOVERY_TIME"
    }

> }'

Reviewer: I think this example will not work because of string interpolation, please confirm exact copy / paste will work

```

</TabItem>

> </Tabs>

Reviewer: since there is only 1 tab, maybe delete this stuff

> * If `recovery_target_time` is omitted from the API call then it will use now as the time

Reviewer: ```suggestion
If `recovery_target_time` is omitted, then it will use now as the time.
```

> For more information to restore from an existing instance, please use the [Tembo Cloud Platform API](https://tembo.io/docs/tembo-cloud/openapi/#tag/instance/operation/restore_instance) documentation.

Reviewer: ```suggestion
For more information to restore from an existing instance, please see the [Tembo Cloud Platform API documentation](https://tembo.io/docs/tembo-cloud/openapi/#tag/instance/operation/restore_instance).
```

- The retention policy for AWS S3 is set to 40 days
- AWS S3 object versioning is enabled, and the retention policy for non-current versions is 30 days

## WAL archiving

Tembo sets `archive_timeout` to 5min, ensuring that WAL files are closed and archived to S3 at least every five minutes, providing a deterministic time-based value for your Recovery Point Objective (RPO).

## Restore

> In Tembo Cloud, we support Point-In-Time Recovery (PITR) to a new Tembo Cluster for the last 30 days. Backup data is saved an additional 30 days after deletion to protect against accidental deletion, so backup data in total is retained for 60 days.

Reviewer: ```suggestion
Tembo Cloud supports Point-In-Time Recovery (PITR) to a new Tembo Instance for all running Instances and for 30 days after deletion.
```

Reviewer: I think we do not mention the extra 30 day under-the-hood retention contingency of previous versions.

:::info
Coming soon: Point-In-Time Recovery (PITR) via the [Tembo Cloud UI](https://cloud.tembo.io)
:::

1. Generate an API token for communicating with your Tembo instance. Navigate to https://cloud.tembo.io/generate-jwt and follow the instructions to generate a token.

2. Set the following environment variables:

    ```bash
    export TEMBO_TOKEN=<your token>
    export TEMBO_ORG=<your organization id>
    export TEMBO_INST=<your instance id>
    ```

3. Initiate a restore from an existing instance using the [Tembo Cloud Platform API](https://tembo.io/docs/tembo-cloud/openapi/#tag/instance/operation/restore_instance):

### Full Restore

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
<TabItem value="curl" label="Curl">

> ```shell

Reviewer: In this section, instead just include all code blocks into a single code block with the variables up top, and right before the code block, link to the authentication guide. Also, I noticed config guide is not doing that, we should update that to also refer to the auth guide and make sure the entire code block is copy / pastable in a single copy, then substitution or org id and so on.

Reviewer: like this 

For information on authenticating to the API, please see the [Tembo Cloud API authentication guide](/docs/tembo-cloud/api-authentication).

curl -X 'POST' \
  "https://api.tembo.io/api/v1/orgs/$TEMBO_ORG/restore" \
  -H "accept: application/json" \
  -H "Authorization: Bearer $TEMBO_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{

>     "instance_name": "$TEMBO_INST-restore",

Reviewer: I think you should define two environment variables for instance to be restored from and instance to restore to, instead of using naming convention

    "restore": {
      "instance_id": "$TEMBO_INST"
    }
}'
```

</TabItem>
</Tabs>

### Point-In-Time (PITR) Restore

1. To initiate a PITR you will need to get the earliest time to recover using the [Tembo Cloud Platform API](https://tembo.io/docs/tembo-cloud/openapi):

    ```bash
    curl -X 'GET' \
      "https://api.tembo.io/api/v1/orgs/$TEMBO_ORG/instances/$TEMBO_INST" \
      -H "accept: application/json" \
      -H "Authorization: Bearer $TEMBO_TOKEN" \
      -H "Content-Type: application/json" | | jq .first_recoverability_time 
    ```

2. Now pick a time you want to restore to based off your `first_recoverability_time` and the current date and time.

    ```bash
    export TEMBO_RECOVERY_TIME="2023-10-18T23:00:08Z"

    ```

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
<TabItem value="curl" label="Curl">

```shell
curl -X 'POST' \
  "https://api.tembo.io/api/v1/orgs/$TEMBO_ORG/restore" \
  -H "accept: application/json" \
  -H "Authorization: Bearer $TEMBO_TOKEN" \
  -H "Content-Type: application/json" \
  -d '{
    "instance_name": "$TEMBO_INST-restore",

>     "restore": {

Reviewer: I think delete this section and down. Instead, just show in the first example how to generate a timestamp for 5 minutes ago (i.e. 1 total example, 1 total code block in this doc). Then after the code block, you can mention that if the recovery_target_time is omitted, that it will pick now as the time. I think don't mention how to fetch the first recoverable time because we already said it should be for 30 days. At the end of the guide, you can link to the tembo cloud API docs, and say go look there for more information, linking to here https://api.tembo.io/redoc#tag/instance/operation/restore_instance
